src/installer/pkg/sfx/Microsoft.NETCore.App/Microsoft.NETCore.App.Crossgen2.sfxproj

```diff
@@ -23,26 +23,21 @@
   <Target Name="PublishCrossgen"
           BeforeTargets="GetFilesToPackage">
 
-    <!-- Copy System.Private.CoreLib from the coreclr bin directory to the runtime pack directory,
-         as we always need the copy of System.Private.CoreLib that matches exactly with the runtime. -->
-    <Copy SourceFiles="$(CoreCLRArtifactsPath)System.Private.CoreLib.dll"
+    <!-- Copy System.Private.CoreLib from the coreclr bin directory to the runtime pack directory
+         if we're not using Native AOT, as we always need the copy of System.Private.CoreLib that
+         matches exactly with the runtime. -->
+    <Copy Condition="'$(NativeAotSupported)' != 'true'"
+          SourceFiles="$(CoreCLRArtifactsPath)System.Private.CoreLib.dll"
           DestinationFolder="$(MicrosoftNetCoreAppRuntimePackNativeDir)"
           SkipUnchangedFiles="true" />
 
-<MSBuild Projects="$(RepoRoot)src/coreclr/tools/aot/crossgen2/crossgen2.csproj"
-             Targets="Restore"
-             Properties="MSBuildRestoreSessionId=$([System.Guid]::NewGuid())
-              ;_IsPublishing=true
-              ;RuntimeIdentifier=$(PackageRID)
-              ;CoreCLRArtifactsPath=$(CoreCLRArtifactsPath)
-              ;R2ROverridePath=$(MSBuildThisFileDirectory)ReadyToRun.targets" />
+    <MSBuild Projects="$(RepoRoot)src/coreclr/tools/aot/crossgen2/crossgen2_live.csproj"
+                Targets="Restore"
+                Properties="MSBuildRestoreSessionId=$([System.Guid]::NewGuid())" />
 
-    <MSBuild Projects="$(RepoRoot)src/coreclr/tools/aot/crossgen2/crossgen2.csproj"
+    <MSBuild Projects="$(RepoRoot)src/coreclr/tools/aot/crossgen2/crossgen2_live.csproj"
              Targets="Publish;PublishItemsOutputGroup"
-             Properties="_IsPublishing=true
-              ;RuntimeIdentifier=$(PackageRID)
-              ;CoreCLRArtifactsPath=$(CoreCLRArtifactsPath)
-              ;R2ROverridePath=$(MSBuildThisFileDirectory)ReadyToRun.targets">
+             Properties="PublishOnBuild=false">
```

Comment:
This is necessary because otherwise there seems to be some sort of cycle of target dependencies. I didn't spend time debugging it since it didn't seem important.